### DISCUSSION TITLE
Use AI for create text with content

### DISCUSSION THREAD

**Discussion by saeedtahmtan** [monkeytypegame/monkeytype#5188] [Polls]
Isn't it better to have a meaningful text to type instead of a meaningless sentence?

**Comment by fehmer** [comment]
One thing to consider is that using LLM/AI to generate text is very costly in terms of compute power and therefor energy consumption.

> You could say that a single LLM interaction may consume as much power as leaving a low-brightness LED lightbulb on for one hour.
source: https://spectrum.ieee.org/ai-energy-consumption

Monkeytype already has multiple ways to train typing with natural text, such as quotes, the wikipedia funbox and custom texts. 

Given the popularity of Monkeytype I vote for not integrating with any LLM and be a bit more eco friendly.

**Reply by localhost-chetan** [reply]
It really need this feature, cause as of now I'm generating those random paragraphs on chatGPT and then pasting it into monkeytype which is becoming really hectic for me

**Comment by birdguy-cs** [comment]
if this feature were to be added i would say it should be that you provide a api key for whichever ai you are going to use that monkeytype uses.

**Reply by saeedtahmtan** [reply]
Ollama and OpenAi integration seems cool right?

**Comment by Charliee420** [comment]
### Issue: Repetitive Word Frequency in Typing Feature
<img width="1920" height="1128" alt="Screenshot 2026-03-04 071430" src="https://github.com/user-attachments/assets/ee4801c6-43f0-455a-84e2-d9f5327e017f" />
<img width="1920" height="1128" alt="Screenshot 2026-03-04 071219" src="https://github.com/user-attachments/assets/4e75fa8d-3423-4c13-8d92-0c9b372afedb" />
<img width="1920" height="1128" alt="Screenshot 2026-03-04 071308" src="https://github.com/user-attachments/assets/67c43208-9c24-4670-b08e-a927ae1e5f84" />
<img width="1920" height="1128" alt="Screenshot 2026-03-04 071335" src="https://github.com/user-attachments/assets/5706130e-0953-4cbe-abdc-a218b240ab47" />
### Typing module shows repetitive words instead of random distribution.
Please improve word randomization and reduce repetition frequency to improve user experience.

### Use AI for this
